database/errors.go

```diff
@@ -2,12 +2,15 @@ package database
 
 import (
 	"errors"
+	"fmt"
 )
 
-var (
-	// ErrTableNotFound is returned when the targeted table doesn't exist.
-	ErrTableNotFound = errors.New("table not found")
+// ErrTableNotFound is returned when the targeted table doesn't exist.
+func ErrTableNotFound(tableName string) error {	
+	return fmt.Errorf("%w %s", errors.New("Error: no such table: "), tableName)
```

Comment:
I believe the test fails due to this line—each error has a different memory location, so you can’t compare them directly.